devito/ir/iet/scheduler.py

```diff
@@ -122,10 +122,11 @@ def iet_make(clusters):
         for j, offs in ispace.sub_iterators.get(k.dim, []):
             modulo = len(offs)
             for n, o in enumerate(filter_ordered(offs)):
-                name = "%s%d" % (j.name, n)
-                vname = Scalar(name=name, dtype=np.int32)
-                value = (k.dim + o) % modulo
-                uindices.append(UnboundedIndex(vname, value, value, j, j + o))
+                if j.is_Stepping:
```

Comment:
Why? would the test that you added fail without this?

Reply:
I don't think I like the use of ``is_Stepping`` here

Reply:
This modulo instanciations are useless for all non-stepping dimensions.

Reply:
without it:
```
devito @ eas-ce-21203 [mlouboutin3]: python tests/test_dimension.py 
/tmp/devito-xb903407/60f8101685795ba12d376ffc42a1298fd38f5883.c: In function ‘Kernel’:
/tmp/devito-xb903407/60f8101685795ba12d376ffc42a1298fd38f5883.c:29:42: error: expected iteration declaration or initialization before ‘x’
       for (int x = x_m, xsub0 = (x)%(1); x <= x_M; x += 1, xsub0 = (x)%(1))
```

Reply:
I think it's because of this:
```
class ConditionalDimension(DerivedDimension):

    is_NonlinearDerived = True
```

Reply:
which in turn causes this:
```
def build_intervals(stencil):
    """
    ...
    """
    iterators = OrderedDict()
    for i in stencil.dimensions:
        if i.is_NonlinearDerived:
            iterators.setdefault(i.parent, []).append(stencil.entry(i))
```

Reply:
**maybe** dropping the ``is_NonlinearDerived`` flag makes it just work. WARNING: this is just a random thought and I might be completely wrong!

Reply:
Just tried, goes crazy. It takes  the SubDim as separate and writes loop for instead of if